backend/app/token_whitelists/whitelists.py

```diff
@@ -7,42 +7,54 @@
 from celery.utils.log import get_task_logger
 from httpx import AsyncClient, HTTPStatusError, RequestError, Timeout
 
-from .storage_helpers import retrieve_token_whitelist, store_token_whitelist
+from .storage_helpers.storage_helpers import (
+    retrieve_token_whitelist,
+    store_token_whitelist,
+)
 
 COVALENT_KEY = os.getenv("COVALENT_KEY")
 COVALENT_POOLS_URL = "https://api.covalenthq.com/v1/{chain_id}/xy=k/{protocol}/pools/"
 
 
-async def _get_covalent_pair_list(
+async def get_single_covalent_pairs_page(
```

Comment:
A good naming for a generator function is a description of what it produces (yields). I'd add that using plural is good for generators.

Here, `single` and `page` are both confusing and don't describe the output. Let's remove them.